File: internal/model/category.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// Category groups inventory items under a reusable label.
type Category struct {
	ID        string    `gorm:"primaryKey;type:varchar(10)" json:"id"`
	TenantID  string    `gorm:"type:varchar(36);not null;default:'default';uniqueIndex:idx_categories_tenant_name" json:"tenant_id"`
	Name      string    `gorm:"type:varchar(100);not null;uniqueIndex:idx_categories_tenant_name" json:"name"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (category *Category) BeforeCreate(*gorm.DB) error {
	if category.ID == "" {
		id, err := newCategoryID()
		if err != nil {
			return err
		}
		category.ID = id
	}

	if category.TenantID == "" {
		category.TenantID = "default"
	}

	return nil
}

File: web/src/types/category.ts
export interface Category {
  id: string
  tenant_id: string
  name: string
  created_at: string
  updated_at: string
}

export interface CreateCategoryPayload {
  name: string
}

export interface UpdateCategoryPayload {
  name?: string
}

export interface CategoryListResponse {
  categories: Category[]
  total: number
}
